client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -651,19 +656,20 @@ public void remove() {
 
     private class InternalRequest {
         private final Request request;
-        private final Map<String, String> params;
         private final Set<Integer> ignoreErrorCodes;
         private final HttpRequestBase httpRequest;
+        private final Cancellable cancellable;
```

Comment:
Do we need to keep track of this object since it is returned to the user ?

Reply:
yes, I also think it could be the InternalRequest itself to be cancellable or some other class built with a static factory method.

Reply:
we do unless we change the synchronization aspect that I commented on above.